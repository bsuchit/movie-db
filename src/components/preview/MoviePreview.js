import React, { useEffect } from "react";
import PropTypes from 'prop-types';

import 'components/components.scss';
import { Divider, Label, List, Loader, Rating } from "semantic-ui-react";
import _ from "lodash";

function MoviePreview({
    title, episode, director, releaseDate, description, producer, getPoster, poster, ratings, loading, runtime, language
}) {
    useEffect(() => {
        getPoster();
    }, [title]);

    if(loading){
        return <Loader active>Loader</Loader>
    }

    return (
        <div className="column preview" data-testid={`preview_${title}`}>
            <div className="row preview_header">
                { poster && 
                    <img className="poster" src={poster} />
                }
                <div>
                    <h1>{`${episode} - ${title}`}</h1>
                    <div>{`Release Date - ${releaseDate}`}</div>
                    <div>{`Director - ${director}`}</div>
                    <div>{`Producer - ${producer}`}</div>
                    <List className="column">
                        <List.Item><Label>{runtime}</Label></List.Item>
                        <List.Item><Label>{language}</Label></List.Item>
                    </List>
                </div>
            </div>
            <Divider />
            <p><h5>Plot</h5>{description}</p>
            <Divider />
            <div className="average_rating">Average Rating - <Rating defaultRating={5} maxRating={10} /></div>
            <div>
                <List className="row">
                    {
                        ratings && _.map(ratings, rating => (<List.Item key={rating.Source}>
                                <Label>
                                    {rating.Source}
                                    <Label.Detail>{rating.Value}</Label.Detail>
                                </Label>
                            </List.Item>)
                        )
                    }
                </List>
            </div>
        </div>
    )
}

MoviePreview.propTypes = {
    title: PropTypes.string,
    episode: PropTypes.number,
    director: PropTypes.string,
    releaseDate: PropTypes.string,
    description: PropTypes.string,
    producer: PropTypes.string,
    getPoster: PropTypes.func,
    poster: PropTypes.string,
    ratings: PropTypes.array,
    loading: PropTypes.bool,
    runtime: PropTypes.string,
    language: PropTypes.string
};

export default MoviePreview;