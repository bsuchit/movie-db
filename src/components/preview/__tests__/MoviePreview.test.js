import React from "react";
import { render } from "@testing-library/react";

import MoviePreview from "../MoviePreview";

describe('Given FilterPopup', () => {
    const mockGetPoster = jest.fn();

    const props = {
        title: "A New Hope",
        episode: 4,
        director: "Chris Nolan",
        releaseDate: "1977-05-25",
        description: "Plot",
        producer: "Gary Kurtz",
        getPoster: mockGetPoster,
        poster: "https://examplexyz.com/poster.png",
        ratings: [{ Source: 'IMDB' , Value: '8.6/10'}],
        loading: false,
        runtime: "121 min",
        language: "English, Hindi"
    }

    it('should render correctly with given props', () => {
        const { container } = render(<MoviePreview {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should show loader WHEN loading', () => {
        const { container } = render(<MoviePreview {...props} loading />);
        expect(container.firstChild).toMatchSnapshot();
    });

});
