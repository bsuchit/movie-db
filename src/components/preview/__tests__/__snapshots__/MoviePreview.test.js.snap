// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given FilterPopup should render correctly with given props 1`] = `
<div
  class="column preview"
>
  <div
    class="row preview_header"
  >
    <img
      class="poster"
      src="https://examplexyz.com/poster.png"
    />
    <div>
      <h1>
        4 - A New Hope
      </h1>
      <div>
        Release Date - 1977-05-25
      </div>
      <div>
        Director - Chris Nolan
      </div>
      <div>
        Producer - Gary Kurtz
      </div>
      <div
        class="ui list column"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          <div
            class="ui label"
          >
            121 min
          </div>
        </div>
        <div
          class="item"
          role="listitem"
        >
          <div
            class="ui label"
          >
            English, Hindi
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ui divider"
  />
  <p>
    <h5>
      Plot
    </h5>
    Plot
  </p>
  <div
    class="ui divider"
  />
  <div
    class="average_rating"
  >
    Average Rating - 
    <div
      class="ui rating"
      role="radiogroup"
      tabindex="-1"
    >
      <i
        aria-checked="false"
        aria-posinset="1"
        aria-setsize="10"
        class="active icon"
        role="radio"
        tabindex="0"
      />
      <i
        aria-checked="false"
        aria-posinset="2"
        aria-setsize="10"
        class="active icon"
        role="radio"
        tabindex="0"
      />
      <i
        aria-checked="false"
        aria-posinset="3"
        aria-setsize="10"
        class="active icon"
        role="radio"
        tabindex="0"
      />
      <i
        aria-checked="false"
        aria-posinset="4"
        aria-setsize="10"
        class="active icon"
        role="radio"
        tabindex="0"
      />
      <i
        aria-checked="true"
        aria-posinset="5"
        aria-setsize="10"
        class="active icon"
        role="radio"
        tabindex="0"
      />
      <i
        aria-checked="false"
        aria-posinset="6"
        aria-setsize="10"
        class="icon"
        role="radio"
        tabindex="0"
      />
      <i
        aria-checked="false"
        aria-posinset="7"
        aria-setsize="10"
        class="icon"
        role="radio"
        tabindex="0"
      />
      <i
        aria-checked="false"
        aria-posinset="8"
        aria-setsize="10"
        class="icon"
        role="radio"
        tabindex="0"
      />
      <i
        aria-checked="false"
        aria-posinset="9"
        aria-setsize="10"
        class="icon"
        role="radio"
        tabindex="0"
      />
      <i
        aria-checked="false"
        aria-posinset="10"
        aria-setsize="10"
        class="icon"
        role="radio"
        tabindex="0"
      />
    </div>
  </div>
  <div>
    <div
      class="ui list row"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <div
          class="ui label"
        >
          IMDB
          <div
            class="detail"
          >
            8.6/10
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Given FilterPopup should show loader WHEN loading 1`] = `
<div
  class="ui active text loader"
>
  Loader
</div>
`;
